File: csharp/word-count/WordCount.cs
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

public static class WordCount
{
    public static IDictionary<string, int> CountWords(string phrase) =>
        Regex.Matches(phrase, @"\b['\w]+\b").Select(m => m.Value)
            .GroupBy(w => w.ToLowerInvariant()).ToDictionary(g => g.Key, g => g.Count());
}

File: vbnet/word-count/WordCount.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text.RegularExpressions

Public Module WordCount
    Public Function CountWords(ByVal s As String) As IDictionary(Of String, Integer)
        Return Regex.Matches(s, "\b[\w']+\b").Select(Function(m) m.Value).GroupBy(Function(w) w.ToLowerInvariant()).ToDictionary(Function(kv) kv.Key, Function(kv) kv.Count())
    End Function
End Module
